packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -538,7 +540,7 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
 
           // record for HMR import chain analysis
           // make sure to unwrap and normalize away base
-          const hmrUrl = unwrapId(url.replace(base, '/'))
+          const hmrUrl = unwrapId(stripBase(url, base))
```

Comment:
we can use `stripBase` on line 268 as well and can probably remove the `if` statement at that line if we make `stripBase` robust enough